components/camel-bean/src/test/java/org/apache/camel/component/bean/BeanInfoTest.java

```diff
@@ -108,6 +111,28 @@ public String myMethod() {
         assertTrue(info.hasAnyMethodHandlerAnnotation());
     }
 
+    @Test
+    public void testSplitSafeQuotesUnremoveQuoteAfterValidParameterTypeCheck() {
+        String[] out = StringQuoteHelper.splitSafeQuote("'Hello,Camel','Bye,World'", ',', true, true);
+        Iterator<?> it = ObjectHelper.createIterator(out, ",", true);
+        Object parameterValue = it != null && it.hasNext() ? it.next() : null;
+        while (parameterValue != null) {
+            assertTrue(BeanHelper.isValidParameterValue(parameterValue.toString()));
+            parameterValue = it != null && it.hasNext() ? it.next() : null;
+        }
+    }
+
+    @Test
+    public void testSplitSafeQuoteRemoveQuoteAfterValidParameterTypeCheck() {
+        String[] out = StringQuoteHelper.splitSafeQuote("'\"Hello,Camel\"','\"Bye,World\"'", ',');
+        Iterator<?> it = ObjectHelper.createIterator(out, ",", true);
+        Object parameterValue = it != null && it.hasNext() ? it.next() : null;
+        while (parameterValue != null) {
+            assertTrue(BeanHelper.isValidParameterValue(parameterValue.toString()));
+            parameterValue = it != null && it.hasNext() ? it.next() : null;
+        }
+    }
```

Comment:
The tests for `StringQuoteHelper` are [here](https://github.com/apache/camel/blob/main/core/camel-core/src/test/java/org/apache/camel/util/StingQuoteHelperTest.java)

Reply:
You will see in the different tests that it is already possible to get `'Hello'` with single quotes if you use double quotes in your expression like `"'Hello'"`.